components/nodemanager-service/tests/project_filtering_test.go

```diff
@@ -17,6 +17,173 @@ import (
 	"github.com/chef/automate/lib/grpc/auth_context"
 )
 
+func TestReadProjectFilteringIngestedNodes(t *testing.T) {
+	timestamp, err := ptypes.TimestampProto(time.Now())
+	require.NoError(t, err)
+
+	db, err := createPGDB()
+	require.NoError(t, err)
+
+	nodeManager := nodesserver.New(db, nil, "")
+
+	// Adding a manual node
+	mgr1 := manager.NodeManager{Name: "mgr1", Type: "aws-ec2"}
+	mgrID1, err := db.AddNodeManager(&mgr1, "11111111")
+	require.NoError(t, err)
+	defer db.DeleteNodeManager(mgrID1)
+
+	node1 := manager.ManagerNode{Id: "i-1111111", Region: "us-west-2", Host: "Node1"}
+
+	instances := []*manager.ManagerNode{&node1}
+	manualNodeIds := db.AddManagerNodesToDB(instances, mgrID1, "242403433", []*manager.CredentialsByTags{}, "aws-ec2")
+	require.NoError(t, err)
+	defer func() {
+		for _, node := range manualNodeIds {
+			db.DeleteNode(node)
+		}
+	}()
+
+	assert.Equal(t, 1, len(manualNodeIds))
+	manualNodeID := manualNodeIds[0]
+
+	cases := []struct {
+		description     string
+		requestProjects []string
+		nodeProjects    []string
+		isError         bool
+	}{
+		{
+			description:     "Node project matching request's projects",
+			requestProjects: []string{"target_project"},
+			nodeProjects:    []string{"target_project"},
+			isError:         false,
+		},
+		{
+			description:     "Node project not matching request's projects",
+			requestProjects: []string{"missed_target_project"},
+			nodeProjects:    []string{"target_project"},
+			isError:         true,
+		},
+		{
+			description:     "Node has no projects; request's project has unassigned project",
+			requestProjects: []string{authzConstants.UnassignedProjectID},
+			nodeProjects:    []string{},
+			isError:         false,
+		},
+		{
+			description:     "Node has a project assigned; request's project has unassigned and the matching project",
+			requestProjects: []string{authzConstants.UnassignedProjectID, "target_project"},
+			nodeProjects:    []string{"target_project"},
+			isError:         false,
+		},
+		{
+			description:     "Node has a project assigned; request's projects has only the unassigned project",
+			requestProjects: []string{authzConstants.UnassignedProjectID},
+			nodeProjects:    []string{"target_project"},
+			isError:         true,
+		},
+		{
+			description:     "Node is assigned a project; request for all projects",
+			requestProjects: []string{authzConstants.AllProjectsExternalID},
+			nodeProjects:    []string{"target_project"},
+			isError:         false,
+		},
+		{
+			description:     "Node has no projects; requested for all projects",
+			requestProjects: []string{authzConstants.AllProjectsExternalID},
+			nodeProjects:    []string{},
+			isError:         false,
+		},
+		{
+			description:     "Node has no projects; requested projects is empty",
+			requestProjects: []string{},
```

Comment:
I'm not sure if it should be possible for the domain to ever receive a request with empty projects 🤔I think authz-service should 403 if there are no allowed projects. maybe there should be an error if the request projects are empty

Reply:
Made an issue to clean this up in all the domains!

https://github.com/chef/automate/issues/2369